RFC-0004-sparse-fill-value/README.md

```diff
@@ -0,0 +1,509 @@
+    equal) that must be taken into account when implementing Autograd
+    backward methods for functions that receive sparse tensors as
+    inputs.
+
+    Sparse tensors with undefined fill-value don't have the intrinsic
+    constraints as discussed above.
+
+
+### Future extensions and existing issues
+
+11. For Graph domain, the undefined fill-value can be specified as a
+    tensor with zero dimension(s) that satisfies all the relations
+    listed above except point 5. Invalidation of the point 5 will
+    provide a consistent way to differentiate between defined and
+    undefined fill-values.
+
+    For instance, to reset a fill-value of a sparse tensor to undefined
+    fill-value, one can use:
+
+    ```python
+    A.fill_value().resize_((0,) * len(A.values().shape[1:]))
+    ```
+
+    Note that this operation is reversible, that is, to reset a
+    undefined fill-value to a defined value, say `1.2`, one can use:
+
+    ```python
+    A.fill_value().resize_(A.values().shape[1:]).fill_(1.2)
+    ```
+
+    This works only when the initial fill-value of `A` is a defined
+    value with the shape constraints specified in point 5.
+
+12. For Statistics domain, where unspecified elements are interpreted
+    as "missing data". The missing data concept is different from
+    undefined value in the sense that it represents a certain
+    measurement flaw that cannot be foreseen to appear in a data while
+    the undefined value represents a structural lack of data that is
+    determined by the system under study.
+
+    The fill-value corresponding to missing data can be specified as
+    `nan` (or a tensor of `nan`-s in case of hybrid tensor). This
+    choice is consistent with convention used in Python Pandas
+    package.
+
+13. The introduction of non-zero fill-value feature requires
+    revisiting the existing sparse tensor API.
+
+    1. The acronym NNZ means the "Number of Non-Zeros" in a sparse
+       tensor. In PyTorch, this acronym is used in several places:
+
+       - in the implementations of sparse tensor and related
+         functionality,
+       - in the `repr` output of COO sparse tensor, see
+         [here](https://pytorch.org/docs/master/generated/torch.sparse_coo_tensor.html#torch.sparse_coo_tensor),
+       - as a private method `_nnz()`, see
+         [here](https://pytorch.org/docs/master/sparse.html?highlight=nnz#torch.sparse.FloatTensor._nnz),
+       - as a optional keyword argument `check_sparse_nnz` in
+         [torch.autograd.gradcheck](https://pytorch.org/docs/master/autograd.html#numerical-gradient-checking).
+
+       The acronym NNZ is misused in PyTorch:
+
+       - `nnz` holds the value of the "Number of Specified Elements"
+         (NSE) in a sparse tensor
+       - `nnz` is not always equal to the number of zeros in the
+         sparse tensor, for instance, the `values` of the sparse
+         tensor in COO format may contain zero values that are not
+         accounted in `nnz`
```

Comment:
This is a good point. +1 for `nse` rather than `nnz`.

Reply:
Made the rename nnz->nse as a recommendation.